File: solutions/swift/robot-name/1/Sources/RobotName/RobotName.swift
class Robot {
  private(set) var name = ""
  init() { resetName() }
  static var l: String { String("ABCDEFGHIJKLMNOPQRSTUVWXYZ".randomElement()!) }
  func resetName() { name = "\(Self.l)\(Self.l)\(Int.random(in: 100...999))" }
}

File: solutions/clojure/robot-name/2/src/robot_name.clj
(ns robot-name)

(def ^:private all-chars "ABCDEFGHIJKLMNOPQRSTUVWXYZ")
(def ^:private all-used (atom #{}))
(defn- gen-name []
  (let [name (format "%s%s%03d"
                     (rand-nth all-chars)
                     (rand-nth all-chars)
                     (rand-int 1000))
        [prev-used] (swap-vals! all-used conj name)]
    (if (prev-used name) (recur) name)))
(defn robot [] (atom (gen-name)))
(defn robot-name [robot] @robot)
(defn reset-name [robot] (reset! robot (gen-name)))
